### DISCUSSION TITLE
How did you fix the battery polarity ?

### DISCUSSION THREAD

**Discussion by daveclifford** [DCC-EX/EX-T3-WiFi#55] [General]
Alex,

If you face the JST connector on the FireeBeetle2 microporcessor board, the + (positive) is on the left and the top has a slot. 

My battery (605080) has the  wires reversed:

---^---  (facing you)
P       N

P = Red 
N = Black

(plus/minux characters don't work in this discussion -  they come out as a degree character and a black dot and spaces are compressed)

 See [https://community.particle.io/t/why-isnt-battery-polarity-always-the-same-in-jst-pins/44474]

Particle, Sparkfun and Adafruit use the same polarity on the JST-PH connector.
On the battery side, with the key facing up, and the holes facing you, the right side is +.

=== RC vs IoT ===
Usually these trends below are found (but don’t assume “Always”):

– RC
Looking at load side (female shell with male pins)
Key/slot up
Red (+) is on the right

**– IoT (and many other electronic industries)
Looking at load side (female shell with male pins)
Key/slot up
Black (-) is on the right**

**(This is the FireBeetle2 Orientation)** 

How did you swap the wires ?

Dave

**Comment by alex-code** [comment]
The batteries I used had the correct polarity already but I found not all batteries are created equal, hence the warnings I include.

This may help https://m.youtube.com/watch?v=0G7iIwfuaJ8

**Reply by daveclifford** [reply]
Today I looked at your YouTube video  and switched the BLACK and RED wires.

1.  I measured the battery voltage - was 3.9v.
2. I connected the battery to the FireBeetle2 and then plugged in the USB-C cable. The previous blinking LED is now a solid RED (with the battery attached).
3. I asked for a [linux lsusb] verbose command summary of the CH340X USB port and got back:
    bmAttributes         0x80       (Bus Powered)
    MaxPower               98mA
4. PlatformIO log is:

del_gzip_data([".pio/build/firebeetle2/spiffs.bin"], ["data.gz"])
Looking for upload port...
Auto-detected: /dev/ttyUSB0
Uploading .pio/build/firebeetle2/spiffs.bin
esptool.py v4.4
Serial port /dev/ttyUSB0
Connecting......................................

**A fatal error occurred: Failed to connect to ESP32: Wrong boot mode detected (0x13)! The chip needs to be in download mode.
For troubleshooting steps visit: https://docs.espressif.com/projects/esptool/en/latest/troubleshooting.html
*** [uploadfs] Error 2**

The journey continues..............................

**Reply by daveclifford** [reply]
The isse was a bad USB socket - closed
